storage/replica_trigger.go

```diff
@@ -290,68 +290,91 @@ func (r *Replica) leasePostCommitTrigger(
 	replicaID roachpb.ReplicaID,
 	prevLease *roachpb.Lease, // TODO(tschottdorf): could this not be nil?
 ) {
-	if prevLease.Replica.StoreID != trigger.lease.Replica.StoreID {
-		// The lease is changing hands. Is this replica the new lease holder?
-		if trigger.lease.Replica.ReplicaID == replicaID {
-			// If this replica is a new holder of the lease, update the low water
-			// mark of the timestamp cache. Note that clock offset scenarios are
-			// handled via a stasis period inherent in the lease which is documented
-			// in on the Lease struct.
-			//
-			// The introduction of lease transfers implies that the previous lease
-			// may have been shortened and we are now applying a formally overlapping
-			// lease (since the old lease holder has promised not to serve any more
-			// requests, this is kosher). This means that we don't use the old
-			// lease's expiration but instead use the new lease's start to initialize
-			// the timestamp cache low water.
-			log.Infof(ctx, "new range lease %s following %s [physicalTime=%s]",
-				trigger.lease, prevLease, r.store.Clock().PhysicalTime())
-			r.mu.Lock()
-			r.mu.tsCache.SetLowWater(trigger.lease.Start)
-			r.mu.Unlock()
+	iAmTheLeaseHolder := trigger.lease.Replica.ReplicaID == replicaID
```

Comment:
'Come with me if you want to lease'

On Mon, Sep 12, 2016, 17:23 Tamir Duberstein <notifications@github.com>
wrote:

> In storage/replica_trigger.go
> <https://github.com/cockroachdb/cockroach/pull/8834#discussion_r78457095>:
>
> > -			// mark of the timestamp cache. Note that clock offset scenarios are
> > -			// handled via a stasis period inherent in the lease which is documented
> > -			// in on the Lease struct.
> > -			//
> > -			// The introduction of lease transfers implies that the previous lease
> > -			// may have been shortened and we are now applying a formally overlapping
> > -			// lease (since the old lease holder has promised not to serve any more
> > -			// requests, this is kosher). This means that we don't use the old
> > -			// lease's expiration but instead use the new lease's start to initialize
> > -			// the timestamp cache low water.
> > -			log.Infof(ctx, "new range lease %s following %s [physicalTime=%s]",
> > -				trigger.lease, prevLease, r.store.Clock().PhysicalTime())
> > -			r.mu.Lock()
> > -			r.mu.tsCache.SetLowWater(trigger.lease.Start)
> > -			r.mu.Unlock()
> > +	iAmTheLeaseHolder := trigger.lease.Replica.ReplicaID == replicaID
>
> https://www.youtube.com/watch?v=-dJolYw8tnk
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/cockroachdb/cockroach/pull/8834/files/db56ac038261f5793b707d48cce0e994f07d2a21#r78457095>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AE135DBOxNTEg8h9HyQZZ0UgWgHaCf8Eks5qpcIogaJpZM4JtckG>
> .
>
-- 

-- Tobias